src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -2099,12 +2099,6 @@ public override void AfterTransactionBegin(ITransaction tx)
 					Interceptor.AfterTransactionBegin(tx);
 				}
 			}
-
-			if (!_transactionCoordinatorShared)
-				foreach (var dependentSession in ConnectionManager.DependentSessions)
-				{
-					dependentSession.AfterTransactionBegin(tx);
-				}
```

Comment:
It seems that they just forgot to add `interceptor.afterTransactionBegin` after they did a refactoring mentioned on the PR. Here is a fix: https://github.com/hibernate/hibernate-orm/commit/fe488976928eae3753c7fcc7a8231001d28054d0. It fixes only the "parent" session. I think you are right and we need to call "dependent" sessions as well.